persistence-cassandra-3.11/src/main/java/io/stargate/db/cassandra/impl/CassandraPersistence.java

```diff
@@ -110,6 +113,24 @@
   private static final int STARTUP_DELAY_MS =
       Integer.getInteger("stargate.startup_delay_ms", 3 * MigrationManager.MIGRATION_DELAY_IN_MS);
 
+  // SCHEMA_SYNC_GRACE_PERIOD should be longer than MigrationManager.MIGRATION_DELAY_IN_MS to allow
+  // the schema pull tasks to be initiated, plus some time for executing the pull requests plus
+  // some time to merge the responses. By default the pull task timeout is equal to
+  // DatabaseDescriptor.getRpcTimeout() (10 sec) and there are no retries. We assume that the merge
+  // operation should complete within the default MIGRATION_DELAY_IN_MS.
```

Comment:
the described logic seems very complex to test and hard to tweak. Do we need to expose the possibility to set it via public `schema_sync_grace_period_ms`? In what cases we may need to change it?

Reply:
This will only need to be changed in the defaul RPC timeouts and `MIGRATION_DELAY_IN_MS` are changed or if schema merges are very slow for some reason... so not likely.

However, we cannot directly derive the value for this setting from other settings because we do not know how long the merge operation is going to take, which is why I made it configurable.